controllers/awscluster_controller.go

```diff
@@ -167,9 +168,8 @@ func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	log = log.WithValues("cluster", klog.KObj(cluster))
 
-	if capiannotations.IsPaused(cluster, awsCluster) {
-		log.Info("AWSCluster or linked Cluster is marked as paused. Won't reconcile")
-		return reconcile.Result{}, nil
+	if isPaused, conditionChanged, err := paused.EnsurePausedCondition(ctx, r.Client, cluster, cluster); err != nil || isPaused || conditionChanged {
```

Comment:
I might be misreading this, but doesn't this mean that when a machine is unpaused, `conditionChanged` is still set to `true`, and reconciliation will be skipped? Are we expecting it to just requeue in that case?

Apologies if this is answered upstream, just want to be clear on the expected behavior.

Reply:
Good catch, I think that's intentional behaviour: https://github.com/kubernetes-sigs/cluster-api/pull/11284/files#r1800591168

Reply:
If `conditionChanged` then this won't immediately requeue, and we'll have to wait for the next resync period (or one of the other events). 

Do we want it to immediately requeue?

Reply:
I'm not 100% I'm following here, I thought we would effectively requeue. We patch the condition, which would cause a new event? Or am I missing something :)

Reply:
Sorry i was the one missing something, doh. Yep as its being patched then it will cause it to be requeue.

Ignore me :)

Reply:
Ok fab! :)